src/routers/alpha-router/quoters/v2-quoter.ts

```diff
@@ -140,6 +141,13 @@ export class V2Quoter extends BaseQuoter<V2CandidatePools, V2Route> {
     if (gasPriceWei === undefined) {
       throw new Error('GasPriceWei for V2Routes is required to getQuotes');
     }
+    // throw if we have no amounts or if there are different tokens in the amounts
+    if (amounts.length == 0 || !amounts.every((amount) => amount.currency.equals(amounts[0]!.currency))) {
+      throw new Error('Amounts must have at least one amount and must be same token');
```

Comment:
ok to validate and throw, although I don't think this error condition will ever happen, due to the implementation in `getAmountDistribution`